### PR TITLE
[WIP]: Use json-iter for JSON parsing

### PR DESCRIPTION
**Pull request by ash2k**
**What this PR does / why we need it**:
1. Changes `k8s.io/apimachinery/pkg/util/json` to use `github.com/json-iterator/go` instead of  `encoding/json`;
2. Changes all `encoding/json` usages and all direct `github.com/json-iterator/go` usages to `k8s.io/apimachinery/pkg/util/json`

This is to ensure json marshaling and unmarshaling is consistent and always uses the same library which is better performance-wise.

**Special notes for your reviewer**:
Inconsistency discovered while working on https://github.com/kubernetes/kubernetes/pull/62981.

**Release note**:
```release-note
NONE
```
/kind enhancement
/sig api-machinery

### CODE REVIEW

File: pkg/api/testing/unstructured_test.go [hunk 1]
```diff
@@ -17,12 +17,16 @@ limitations under the License.
 package testing
 
 import (
+	encodingjson "encoding/json"
```
[the remark this answers is not in the document]
Comment (ash2k) [reply pkg/api/testing/unstructured_test.go]: Agree. I'm just debugging an issue where something is not identical between two libraries. Trying to come up with a test by no dice so far.

**Comment by ash2k** [comment]
Blocked on https://github.com/json-iterator/go/issues/272

**Comment by thockin** [comment]
Idea seems sound

**Comment by wojtek-t** [comment]
@ash2k - do you still plan to work on that?

**Comment by ash2k** [comment]
@wojtek-t no, don't have any spare bandwidth for this. Feel free to take over, I'll close this PR. The biggest issue with this PR (apart from lack of time) is that jsoniter-go has some bug/incompatibility with encoding/json package. Finding it and fixing it requires even more time.

**Pull request closed without merging**